iFixit/react-commerce
# Build seeding api and interface

closes #739 

## Changelog

1. Added a UI and API to seed data from arbitrary strapi instances 
  <img width="1079" alt="Screenshot 2022-09-26 at 16 03 34" src="https://user-images.githubusercontent.com/4640135/192296861-7a1ec2e8-3955-4a5c-ab32-9a3b0f151ec9.png">
2. Default to import data from `main.govinor.com` instead of `strapi.ifixit.com`

This approach will enable us to customize the data we use for our previews (by editing `main.govinor.com`) while keeping it up to date with production (by using the seeding interface when we want to update `main.govinor.com` ).
Whenever `main.govinor.com` will be reset with production data, changes will need to be re-applied (right now this means just adding back a test store content type which is basically a duplicate of the us store with `ifixit-test` shopify credentials).
I have considered also hard coding the `test` store content type in the seed, but it seemed like a less flexible solution. 

@sterlinghirsh we should update `strapi.ifixit.com` when this gets merged. The UI for seeding shouldn't be available in production. It's only enabled if `STRAPI_ADMIN_ENABLE_ADDONS_DANGEROUS_ACTIONS` is set.

## QA

1. Open [govinor preview for this branch](https://add-seed-from-arbitrary-strapi-instance.govinor.com/admin/plugins/addons)
2. Verify that you can see the new seed UI
3. Visit `Content Manager` > `Store` and delete the stores
4. Go back to the seeding interface and seed with `main.govinor.com`
5. Head back to `Content Manager` > `Store` and verify that stores have been seeded according to main.govinor.com
6. Visit the Vercel preview and verify that the `Test` store is present on the footer region selector
